alert/store.go

```diff
@@ -255,9 +260,63 @@ func NewDB(ctx context.Context, db *sql.DB, logDB alertlog.Store) (*DB, error) {
 			FROM services
 			WHERE id = $1
 		`),
+
+		findPendingNotifications: p(`
+			SELECT
+			COALESCE(ucm.type::text, nc.type::text) AS dest_type,
+			COALESCE(u.name, nc.name) AS dest_name
+			FROM outgoing_messages om
+			LEFT JOIN user_contact_methods ucm 
+			ON om.user_id = ucm.user_id
+			LEFT JOIN notification_channels nc 
+			ON nc.id = om.channel_id
+			LEFT JOIN users u
+			ON u.id = om.user_id
+			WHERE om.last_status='pending'
+			AND (now() - om.created_at) > INTERVAL '1' MINUTE
+			AND (om.alert_id = $1 OR (om.message_type = 'alert_notification_bundle' AND om.service_id = $2));
+		`),
 	}, prep.Err
 }
 
+func (db *DB) FindPendingNotifications(ctx context.Context, alertID int, serviceID string) ([]AlertPendingNotification, error) {
+	err := permission.LimitCheckAny(ctx, permission.User)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validate.UUID("ServiceID", serviceID)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := db.findPendingNotifications.QueryContext(ctx, alertID, serviceID)
+	if errors.Is(err, sql.ErrNoRows) {
+		err = nil
```

Comment:
Can we return here since we know there are no rows?

```suggestion
		return nil, nil
```